File: src/main/java/net/sf/rails/game/state/ColorModel.java
package net.sf.rails.game.state;

import java.awt.Color;

/**
 * ColourModel defines colors for UI components
 */

public abstract class ColorModel extends Model {

    protected ColorModel(Item parent, String id) {
        super(parent, id);
    }
    
    public abstract Color getBackground();
    
    public abstract Color getForeground();
    
}

File: dotnet/RailsLib.Net/Net/Game/Model/ColorModel.cs
using GameLib.Net.Game.State;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Text;

/**
 * ColourModel defines colors for UI components
 */


namespace GameLib.Net.Game.Model
{
    abstract public class ColorModel : State.Model
    {
        protected ColorModel(IItem parent, string id) : base(parent, id)
        {
            
        }

        public abstract Color Background { get; }

        public abstract Color Foreground { get; }
    }
}
